core/state_processor.go

```diff
@@ -133,10 +158,314 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	}, nil
 }
 
+// ProcessWithAccessList performs EVM execution and state root computation for a block which is known
+// to contain an access list.
+func (p *StateProcessor) ProcessWithAccessList(block *types.Block, statedb *state.StateDB, cfg vm.Config) (chan *ProcessResultWithMetrics, error) {
+	var (
+		header      = block.Header()
+		blockHash   = block.Hash()
+		blockNumber = block.Number()
+		resCh       = make(chan *ProcessResultWithMetrics)
+		requests    [][]byte
+		signer      = types.MakeSigner(p.config, header.Number, header.Time)
+		ctx, cancel = context2.WithCancel(context2.Background())
+	)
+
+	type txExecResult struct {
+		idx     int
+		receipt *types.Receipt
+		err     error
+	}
+
+	txResCh := make(chan txExecResult)
+	rootCalcErrCh := make(chan error) // used for communicating if the state root calculation doesn't match the reported root
+	pStart := time.Now()
+	var (
+		tPreprocess           time.Duration // time to create a set of prestates for parallel transaction execution
+		tPreprocessStateDiffs time.Duration // time that it took to build state diffs from the BAL
+		tVerifyStart          time.Time
+		tVerify               time.Duration // time to compute and verify the state root
+		tExecStart            time.Time
+		tExec                 time.Duration // time to execute block transactions
+		tPostprocess          time.Duration // time to perform post-transaction execution system calls and withdrawals.
+		tPreprocessLoad       time.Duration
+		modifiedPrestate      = make(map[common.Address]*types.StateAccount)
+	)
+
+	// called by resultHandler when all transactions have successfully executed.
+	// performs post-tx state transition (system contracts and withdrawals)
+	// and calculates the ProcessResult, returning it to be sent on resCh
+	// by resultHandler
+	prepareExecResult := func(postTxState *state.StateDB, receipts types.Receipts) *ProcessResult {
```

Comment:
It's fairly hard to follow the logic with these larger function variables. I think it's reasonable to separate this logic into separate methods or create a new object the focuses solely on validating blocks with access lists.